test/langtools/jdk/javadoc/doclet/testConstantValuesPage/TestConstantValuesPage.java

```diff
@@ -41,14 +43,215 @@ public static void main(String... args) throws Exception {
         tester.runTests();
     }
 
+    ToolBox tb = new ToolBox();
+
+    /**
+     * Test to make sure that constant values page does not get
+     * generated when doclet has nothing to document.
+     */
     @Test
-    public void test() {
+    public void testNoPage() {
         javadoc("-d", "out",
                 "-sourcepath", testSrc,
                 "foo");
         checkExit(Exit.CMDERR);
 
         checkOutput(Output.OUT, false,
                 "constant-values.html...");
+        checkFiles(false, "constant-values.html");
+    }
+
+    /**
+     * Tests the "contents" list for a group of named packages in the unnamed module.
+     */
+    @Test
+    public void testIndexNamed(Path base) throws Exception {
+        Path src = base.resolve("src");
+        tb.writeJavaFiles(src,
+                """
+                    package p1.p2a.p3a;
+                    public class CA {
+                        public static final int ia = 1;
+                        public static final String sa = "string";
+                    }
+                    """,
+                """
+                    package p1.p2a.p3b;
+                    public class CB {
+                        public static final int ib = 1;
+                        public static final String sb = "string";
+                    }
+                    """,
+                """
+                    package p1.p2b.p3c;
+                    public class CC {
+                        public static final int ic = 1;
+                        public static final String sc = "string";
+                    }
+                    """,
+                """
+                    package p2;
```

Comment:
yes, the constants page should only include the "included packages" given on the command line.

Reply:
I was asking because the test will still pass if that fourth (trailing) text block is deleted. So I wasn't sure what that text block was -- a part of the test or an artefact.